core/util.go

```diff
@@ -95,6 +96,19 @@ func Fingerprint256(data []byte) string {
 	return B64enc(d.Sum(nil))
 }
 
+func KeyDigest(key crypto.PublicKey) string {
+	switch t := key.(type) {
+		case *jose.JsonWebKey:
+			return KeyDigest(t.Key)
+		case jose.JsonWebKey:
+			return KeyDigest(t.Key)
+		default:
+			keyDER, _ := x509.MarshalPKIXPublicKey(key)
```

Comment:
I worry a little about ignoring this error since we're using default here.

If something weird gets past Jose, it will still get hashed even if it's not a key. I haven't dug into `jose.ParseSigned()` yet to see how likely that is to occur.

As is, if type cannot be unmarshalled as a PKIX public key, it will always yield the same digest `47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=`. See here:

https://play.golang.org/p/V5YRABSrZy